packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -455,3 +408,122 @@ function defaultGenerateClientInfo({ request }: GraphQLRequestContext) {
     return {};
   }
 }
+
+// Creates trace details from request variables, given a specification for modifying
+// values of private or sensitive variables.
+// The details include the variables in the request, TODO(helen): and the modifier type.
+// If sendVariableValues is {safelistAll: bool} or {exceptVariableNames: Array}, it will act similarly to
+// to the to-be-deprecated options.privateVariables, except that the redacted variable
+// names will still be visible in the UI when 'true.'
+// If sendVariableValues is null, we default to the safeListAll = false case.
+export function makeTraceDetails(
+  variables: Record<string, any>,
+  sendVariableValues?: VariableValueOptions,
+  operationString?: string,
+): Trace.Details {
+  const details = new Trace.Details();
+  const variablesToRecord = (() => {
+    if (sendVariableValues && 'transform' in sendVariableValues) {
+      // Custom function to allow user to specify what variablesJson will look like
+      const originalKeys = Object.keys(variables);
+      const modifiedVariables = sendVariableValues.transform({
+        variables: variables,
+        operationString: operationString,
+      });
+      return cleanModifiedVariables(originalKeys, modifiedVariables);
+    } else {
+      return variables;
+    }
+  })();
+
+  // Note: we explicitly do *not* include the details.rawQuery field. The
+  // Engine web app currently does nothing with this other than store it in
+  // the database and offer it up via its GraphQL API, and sending it means
+  // that using calculateSignature to hide sensitive data in the query
+  // string is ineffective.
+  Object.keys(variablesToRecord).forEach(name => {
+    if (
+      !sendVariableValues ||
+      'sendNone' in sendVariableValues ||
```

Comment:
Maybe `sendVariableValues.sendNone` instead. If the user is using TypeScript then the conditions are equivalent, but if not, they could type `sendVariableValues: {sendNone: false}` or something, and while that is a violation of our API, it's still kinda surprising if that's the same as true?

Reply:
Makes sense! I'm going to add a check for { 'sendAll': false } too.